whois-nrtm/src/main/java/net/ripe/db/whois/nrtm/BaseNrtmServerPipelineFactory.java

```diff
@@ -1,86 +0,0 @@
-package net.ripe.db.whois.nrtm;
-
-import net.ripe.db.whois.common.pipeline.MaintenanceHandler;
-import org.jboss.netty.buffer.ChannelBuffer;
-import org.jboss.netty.buffer.ChannelBuffers;
-import org.jboss.netty.channel.ChannelPipeline;
-import org.jboss.netty.channel.ChannelPipelineFactory;
-import org.jboss.netty.channel.Channels;
-import org.jboss.netty.handler.codec.frame.DelimiterBasedFrameDecoder;
-import org.jboss.netty.handler.codec.string.StringDecoder;
-import org.jboss.netty.handler.codec.string.StringEncoder;
-import org.jboss.netty.handler.execution.ExecutionHandler;
-import org.jboss.netty.handler.execution.OrderedMemoryAwareThreadPoolExecutor;
-
-import java.nio.charset.StandardCharsets;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-
-
-abstract class BaseNrtmServerPipelineFactory implements ChannelPipelineFactory {
-
-    private static final ChannelBuffer LINE_DELIMITER = ChannelBuffers.wrappedBuffer(new byte[]{'\n'});
-
-    private static final long MEMORY_SIZE_UNLIMITED = 0;
-    private static final long TIMEOUT_SECONDS = 60L;
-    private static final int POOL_SIZE = 32;
-
-    private final StringDecoder stringDecoder = new StringDecoder(StandardCharsets.UTF_8);
-    private final StringEncoder stringEncoder = new StringEncoder(StandardCharsets.UTF_8);
-
-    protected final ExecutionHandler executionHandler = new ExecutionHandler(
```

Comment:
`ExecutionHandler` has been removed in Netty 4 and replaced with `EventExecutor`. See [No more ExecutionHandler - it's in the core](https://netty.io/wiki/new-and-noteworthy-in-4.0.html#no-more-executionhandler---its-in-the-core).

Depending on the result of the load testing, the following implementations of `ExecutionHandler` _might_ be configured and used:

1. [DefaultEventExecutor](https://netty.io/4.1/api/io/netty/util/concurrent/DefaultEventExecutor.html). This might require having to use those constructors which allows provision of implementation of [ThreadFactory](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadFactory.html) and [Executor](https://docs.oracle.com/javase/10/docs/api/java/util/concurrent/Executor.html) 
2. [ImmediateEventExecutor](https://netty.io/4.1/api/io/netty/util/concurrent/ImmediateEventExecutor.html)

Reply:
I'm testing using an EventExecutorGroup to avoid blocking the Netty I/O thread.

See https://livebook.manning.com/book/netty-in-action/chapter-7/1